File: packages/core/tests/fixtures/holdout/healthy/date_utils.go
package utils

import "time"

// DaysBetween returns the number of days between start and end.
func DaysBetween(start, end time.Time) int {
	diff := end.Sub(start)
	return int(diff.Hours() / 24)
}

// AddDays returns a new time.Time that is days after t.
func AddDays(t time.Time, days int) time.Time {
	return t.AddDate(0, 0, days)
}

// IsWeekend reports whether t falls on a Saturday or Sunday.
func IsWeekend(t time.Time) bool {
	wd := t.Weekday()
	return wd == time.Saturday || wd == time.Sunday
}

// FormatISO returns an ISO-8601 date string (YYYY-MM-DD) for t.
func FormatISO(t time.Time) string {
	return t.Format("2006-01-02")
}

// IsLeapYear reports whether the year of t is a leap year.
func IsLeapYear(t time.Time) bool {
	y := t.Year()
	return (y%4 == 0 && y%100 != 0) || y%400 == 0
}

File: packages/core/tests/fixtures/holdout/healthy/date_utils.py
"""Date utility functions — simple helpers using the standard library only."""
from datetime import date, timedelta
from typing import Optional


def days_between(start: date, end: date) -> int:
    """Return the number of days between two dates (end − start)."""
    return (end - start).days


def add_days(d: date, days: int) -> date:
    """Return the date that is `days` after the given date."""
    return d + timedelta(days=days)


def is_weekday(d: date) -> bool:
    """Return True when d falls on a Monday–Friday."""
    return d.weekday() < 5


def format_iso(d: date) -> str:
    """Return the ISO-8601 string representation of a date (YYYY-MM-DD)."""
    return d.isoformat()


def parse_iso(text: str) -> Optional[date]:
    """Parse an ISO-8601 date string; return None on invalid input."""
    try:
        return date.fromisoformat(text)
    except ValueError:
        return None
